web/source/osk/banner.ts

```diff
@@ -301,7 +328,87 @@ namespace com.keyman.osk {
     }
   }
 
-  class SuggestionTouchManager extends dom.UITouchHandlerBase<HTMLDivElement> {
+  /**
+   * Function     SuggestionBanner
+   * Scope        Public
+   * @param {number} height - If provided, the height of the banner in pixels
+   * Description  Display lexical model suggestions in the banner
+   */
+  export class SuggestionBanner extends Banner {
+    public static readonly SUGGESTION_LIMIT: number = 3;
+    public static readonly MARGIN = 1;
+
+    private options : BannerSuggestion[];
+
+    private manager: SuggestionManager;
+
+    static readonly TOUCHED_CLASS: string = 'kmw-suggest-touched';
+
+    constructor(height?: number) {
+      super(height || SuggestionBanner.DEFAULT_HEIGHT);
```

Comment:
`SuggestionBanner.DEFAULT_HEIGHT` - I find this a little bit confusing since `SuggestionBanner` doesn't directly define `DEFAULT_HEIGHT`. Might be better to change it to `Banner.DEFAULT_HEIGHT` or maybe move the definition to this class in case it's used nowhere else.